### PR TITLE
dev-python/numpy: bump version to 1.18.2

### PR DESCRIPTION
**Pull request by vpayno**
@zmedico @gyakovlev @chutz

Tested with and without lapack with Python 3.6, 3.7 and 3.8.

```
===== 10423 passed, 77 skipped, 18 xfailed, 3 xpassed in 418.24s (0:06:58) =====
===== 10416 passed, 84 skipped, 18 xfailed, 3 xpassed in 385.31s (0:06:25) =====
===== 10426 passed, 76 skipped, 18 xfailed, 3 xpassed in 381.29s (0:06:21) =====
```

### CODE REVIEW

File: dev-python/numpy/numpy-1.18.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	sed -i -e 's:test_large_zip:_&:' numpy/lib/tests/test_io.py || die
+
+	distutils-r1_python_prepare_all
+}
+
+python_compile() {
+	export MAKEOPTS=-j1 #660754
+
+	local python_makeopts_jobs=""
+	python_is_python3 || python_makeopts_jobs="-j $(makeopts_jobs)"
```
Comment (chutz) [inline dev-python/numpy/numpy-1.18.2.ebuild:118]: This line is no longer needed as the ebuild dies not support python2

File: dev-python/numpy/numpy-1.18.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	fi
+
+	# only one fortran to link with:
+	# linking with cblas and lapack library will force
+	# autodetecting and linking to all available fortran compilers
+	append-fflags -fPIC
+	if use lapack; then
+		NUMPY_FCONFIG="config_fc --noopt --noarch"
+		# workaround bug 335908
+		[[ $(tc-getFC) == *gfortran* ]] && NUMPY_FCONFIG+=" --fcompiler=gnu95"
```
Comment (chutz) [inline dev-python/numpy/numpy-1.18.2.ebuild:99]: It looks like this is using gfortran, but the dependencies don't pull it in, there probably needs to be something like `lapack? ( sys-devel/gcc[fortran] )` in `BDEPEND`

File: dev-python/numpy/numpy-1.18.2.ebuild (hunk 1 above)
Comment (vpayno) [reply dev-python/numpy/numpy-1.18.2.ebuild]: removed

File: dev-python/numpy/numpy-1.18.2.ebuild (hunk 2 above)
Comment (vpayno) [reply dev-python/numpy/numpy-1.18.2.ebuild:103]: added

**Comment by vpayno** [comment]
```
--- numpy-1.17.4-r2.ebuild      2020-04-10 15:59:58.417302794 -0700
+++ numpy-1.18.2.ebuild 2020-04-11 21:14:42.835954253 -0700
@@ -8,7 +8,7 @@
  
 FORTRAN_NEEDED=lapack
                                                                                              
-inherit distutils-r1 flag-o-matic fortran-2 multiprocessing toolchain-funcs
+inherit autotools distutils-r1 flag-o-matic fortran-2 multiprocessing toolchain-funcs
                              
 DOC_PV="1.16.4"
 DESCRIPTION="Fast array and numerical python library"
@@ -23,26 +23,37 @@
 LICENSE="BSD"          
 SLOT="0"         
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc-solaris ~x64-solaris ~x86-solaris"
-IUSE="doc lapack test"
-RESTRICT="!test? ( test )"
+IUSE="doc lapack"
               
 RDEPEND="       
        lapack? (
                >=virtual/cblas-3.8
                >=virtual/lapack-3.8
        )"                 
-DEPEND="${RDEPEND}"                                                                          
-BDEPEND="app-arch/unzip
-       dev-python/setuptools[${PYTHON_USEDEP}]
-       lapack? ( virtual/pkgconfig )
+DEPEND="${RDEPEND}                  
        test? (                                                                               
-               dev-python/pytest[${PYTHON_USEDEP}]
-       )"                               
+               >=dev-python/cython-0.29.15[${PYTHON_USEDEP}]
+               >=dev-python/pytest-cov-2.8.1[${PYTHON_USEDEP}]
+               >=dev-python/hypothesis-5.8.0[${PYTHON_USEDEP}]
+               >=dev-python/pytz-2019.3[${PYTHON_USEDEP}]
+               >=dev-python/cffi-1.14.0[${PYTHON_USEDEP}]
+       )                        
+"                                                                                            
+BDEPEND="
+       app-arch/unzip  
+       lapack? (                                                                             
+               >=sys-devel/automake-1.16.2
+               sys-devel/gcc[fortran]                                                        
+               virtual/pkgconfig                                                             
+       )
+"                                  
                                                                                                                                                                                            
 PATCHES=(       
        "${FILESDIR}"/${PN}-1.17.4-no-hardcode-blasv2.patch
 )
                                                                                                                                                                                            
+distutils_enable_tests pytest                                                                                                                                                              
+                                              
 src_unpack() {
        default      
        if use doc; then
@@ -50,6 +61,11 @@                                                                                                                                                                          
        fi
 }
  
+src_prepare() {
+       default
+       eautomake
+}
+
 python_prepare_all() {
        if use lapack; then
                local incdir="${EPREFIX}"/usr/include
@@ -104,7 +120,6 @@
        export MAKEOPTS=-j1 #660754
  
        local python_makeopts_jobs=""
-       python_is_python3 || python_makeopts_jobs="-j $(makeopts_jobs)"
        distutils-r1_python_compile \
                ${python_makeopts_jobs} \
                ${NUMPY_FCONFIG}
@@ -128,7 +143,7 @@
 }
  
 python_install_all() {
-       local DOCS=( THANKS.txt )
+       local DOCS=( LICENSE.txt README.md THANKS.txt )
  
        if use doc; then
                local HTML_DOCS=( "${WORKDIR}"/html/. )
```

File: dev-python/numpy/numpy-1.18.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	cd "${TMPDIR}" || die
+
+	"${EPYTHON}" -c "
+import numpy, sys
+r = numpy.test(label='full', verbose=3)
+sys.exit(0 if r else 1)" || die "Tests fail with ${EPYTHON}"
+}
+
+python_install() {
+	distutils-r1_python_install ${NUMPY_FCONFIG}
```
Comment (SoapGentoo) [inline dev-python/numpy/numpy-1.18.3.ebuild:142]: this is missing the fix for missing files.

Comment (vpayno) [reply dev-python/numpy/numpy-1.18.3.ebuild:144]: Added the fix from https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=1b53d722d793c11f16e81bc2c468cbb74d0c8a32 .

**Comment by heroxbd** [comment]
@patrick-mclean Thanks for helping us maintain numpy.  However, this PR introduces 2 bugs and 1 issue

Bug: https://bugs.gentoo.org/660754
Bug: https://bugs.gentoo.org/700848

1 issue: massive drop of keywords.  Have you ever run repoman against user submitted ebuilds?

File: dev-python/numpy/numpy-1.18.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	default
+	if use doc; then
+		unzip -qo "${DISTDIR}"/numpy-html-${DOC_PV}.zip -d html || die
+	fi
+}
+
+src_prepare() {
+	default
+	eautomake
+}
```
Comment (heroxbd) [inline dev-python/numpy/numpy-1.18.3.ebuild:68]: This skips `python_prepare_all`.

File: dev-python/numpy/numpy-1.18.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+SRC_URI="
+	mirror://pypi/${PN:0:1}/${PN}/${P}.zip
+	doc? (
+		https://numpy.org/doc/$(ver_cut 1-2 ${DOC_PV})/numpy-html.zip -> numpy-html-${DOC_PV}.zip
+		https://numpy.org/doc/$(ver_cut 1-2 ${DOC_PV})/numpy-ref.pdf -> numpy-ref-${DOC_PV}.pdf
+		https://numpy.org/doc/$(ver_cut 1-2 ${DOC_PV})/numpy-user.pdf -> numpy-user-${DOC_PV}.pdf
+	)"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~x86"
```
Comment (heroxbd) [inline dev-python/numpy/numpy-1.18.3.ebuild:27]: Why do you drop all the other keywords?

File: dev-python/numpy/numpy-1.18.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+	find numpy/*/tests -name '*.py' -exec sed -i \
+		-e 's:def \(.*_fuzz\):def _\1:' {} + || die
+	# very memory- and disk-hungry
+	sed -i -e 's:test_large_zip:_&:' numpy/lib/tests/test_io.py || die
+
+	distutils-r1_python_prepare_all
+}
+
+python_compile() {
+	distutils-r1_python_compile ${NUMPY_FCONFIG}
```
Comment (heroxbd) [inline dev-python/numpy/numpy-1.18.3.ebuild:121]: You have overridden the fix committed in https://bugs.gentoo.org/660754, and this causes a regression now.  Is there a reason?

File: dev-python/numpy/numpy-1.18.3.ebuild (hunk 5 above)
Comment (vpayno) [reply dev-python/numpy/numpy-1.18.3.ebuild:27]: https://bugs.gentoo.org/721130

File: dev-python/numpy/numpy-1.18.3.ebuild (hunk 4 above)
Comment (vpayno) [reply dev-python/numpy/numpy-1.18.3.ebuild:68]: I can't figure out why I added that. I couldn't find it in other older numpy ebuilds, I must have accidentally copied it from a different ebuild.

File: dev-python/numpy/numpy-1.18.3.ebuild (hunk 6 above)
Comment (vpayno) [reply dev-python/numpy/numpy-1.18.3.ebuild:121]: I was asked to see if it worked without it. We should have left it alone.

**Comment by vpayno** [comment]
https://bugs.gentoo.org/660754: Already marked resolved.

https://bugs.gentoo.org/700848: Already marked resolved.

Created this bug to get the keywords updated: https://bugs.gentoo.org/721130

File: dev-python/numpy/numpy-1.18.3.ebuild (hunk 5 above)
Comment (heroxbd) [reply dev-python/numpy/numpy-1.18.3.ebuild:27]: It's fine to drop keywords if the dependencies cannot be met.  But please document the reasons in the comment messages.

File: dev-python/numpy/numpy-1.18.3.ebuild (hunk 4 above)
Comment (heroxbd) [reply dev-python/numpy/numpy-1.18.3.ebuild:68]: Okay.  No problem.

File: dev-python/numpy/numpy-1.18.3.ebuild (hunk 6 above)
Comment (heroxbd) [reply dev-python/numpy/numpy-1.18.3.ebuild:121]: You could dig through the git history to see why the hack was introduced. 

In the ebuild, it said

`export MAKEOPTS=-j1 #660754`

It makes sense to study the bug carefully before removing it.

**Pull request closed without merging**